src/darwin/Framework/CHIP/CHIPThreadOperationalDataset.mm

```
@@ -0,0 +1,114 @@
+        _masterKey = masterKey;
+        _PSKc = PSKc;
+        _channel = channel;
+        _panID = panID;
+        _cppThreadOperationalDataset = chip::Thread::OperationalDataset();
+        if ([self _populateCppOperationalDataset]) {
+            return self;
+        }
+    }
+    return nil;
+}
+
+- (BOOL)_populateCppOperationalDataset
+{
+    _cppThreadOperationalDataset.Clear();
+    _cppThreadOperationalDataset.SetNetworkName([self.networkName cStringUsingEncoding:NSUTF8StringEncoding]);
+
+    if (![self _checkDataLength:self.extendedPANID expectedLength:chip::Thread::kSizeExtendedPanId]) {
+        CHIP_LOG_ERROR("Invalid ExtendedPANID");
+        return NO;
+    }
+    uint8_t extendedPanId[chip::Thread::kSizeExtendedPanId];
+    [self.extendedPANID getBytes:&extendedPanId length:chip::Thread::kSizeExtendedPanId];
+    _cppThreadOperationalDataset.SetExtendedPanId(extendedPanId);
+
+    if (![self _checkDataLength:self.masterKey expectedLength:chip::Thread::kSizeMasterKey]) {
+        CHIP_LOG_ERROR("Invalid MasterKey");
+        return NO;
+    }
+    uint8_t masterKey[chip::Thread::kSizeMasterKey];
+    [self.masterKey getBytes:&masterKey length:chip::Thread::kSizeMasterKey];
+    _cppThreadOperationalDataset.SetMasterKey(masterKey);
+
+    if (![self _checkDataLength:self.PSKc expectedLength:chip::Thread::kSizePSKc]) {
+        CHIP_LOG_ERROR("Invalid PKSc");
+        return NO;
+    }
+    uint8_t PSKc[chip::Thread::kSizePSKc];
+    [self.PSKc getBytes:&PSKc length:chip::Thread::kSizePSKc];
+    _cppThreadOperationalDataset.SetPSKc(PSKc);
+
+    _cppThreadOperationalDataset.SetChannel(self.channel);
+
+    uint16_t * valuePtr = (uint16_t *) [self.panID bytes];
+    if (valuePtr == nullptr) {
+        return NO;
+    }
+    _cppThreadOperationalDataset.SetPanId(*valuePtr);
+
+    return YES;
+}
+
+- (BOOL)_checkDataLength:(NSData *)data expectedLength:(size_t)expectedLength
+{
+    if (data.length < expectedLength) {
```

Should this perhaps be an equality check?  Or is it really OK to silently drop part of the incoming data if it's too long?